ethclient/ethclient.go

```diff
@@ -198,6 +198,30 @@ func (tx *rpcTransaction) UnmarshalJSON(msg []byte) error {
 	return json.Unmarshal(msg, &tx.txExtraInfo)
 }
 
+// TransactionByHash returns the transaction with the given hash.
+// The code is exactly copied from the function TransactionByHash
+// but this returns the blockHexString instead of json.BlockNumber == nil
+func (ec *Client) TransactionByHashWithBlockNum(ctx context.Context, hash common.Hash) (tx *types.Transaction, blockHexString string, err error) {
```

Comment:
Not sure if we need this. I'll look through for usage and also update the comment to reflect what it actually does if we do

Reply:
Hmm.. ya if this was in a public PR I think it might open a separate can of worms. For Receipts we added block info to the Receipt struct and made changes throughout geth to populate it. For a Transaction there is already an RPC transaction, which may be worth migration to or returning here.

Reply:
Yeah I would anticipate resistance to the TransactionReader interface change - adding interface methods is a breaking change for other implementors.

Potentially-silly question, not knowing the contents of the return structs: A transaction won't be assigned a block num unless it has been mined, right? So we ought to be able to use a combination of the new`TransactionReceipt` (with block info) and `TransactionByHash`to accomplish whatever we were doing with this method?

Reply:
Yeah I pushed back pretty hard when he was trying to add this but I guess he just did it anyway

Reply:
axed!